### PR TITLE
Datafusion year & support for DaskSqlDialect

### LINKED ISSUE #465: [DF] Example of creating a custom function
**Is your feature request related to a problem? Please describe.**
A concise example of creating a custom UDF using the new Datafusion syntax would allow for users to understand writing, registering, and using UDFs that are not supported by Dask-SQL out of the box.

**Describe the solution you'd like**
Straight forward README and code example that lays out how a user can create a custom UDF

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by jdye64**
Closes: #538 

Since `year()` is not part of the standard function list, aka custom, this also closes #465 since an example is provided here.

### CODE REVIEW

File: tests/integration/test_rex.py [hunk 1]
```diff
@@ -16,7 +26,7 @@ def test_case(c, df):
         """
     SELECT
         (CASE WHEN a = 3 THEN 1 END) AS "S1",
-        (CASE WHEN a > 0 THEN a ELSE 1 END) AS "S2",
+        (CASE WHEN a > 0 THEN a ELSE 1 END) AS "S2",df.shape[0]
```
Comment (jdye64) [inline tests/integration/test_rex.py:29]: oops. Added `df.shape[0]` by mistake. Will remove that.

Comment (andygrove) [inline tests/integration/test_rex.py:29]: This seems like it might be an unintentional change?

File: dask_planner/src/expression.rs [hunk 2]
```diff
@@ -301,6 +300,13 @@ impl PyExpr {
                 unimplemented!("ScalarSubquery")
             }
             Expr::IsNotNull(expr) => Ok(vec![PyExpr::from(*expr.clone(), self.input_plan.clone())]),
+            Expr::ScalarUDF { fun: _, args } => {
+                let mut exprs: Vec<PyExpr> = Vec::new();
+                for arg in args {
+                    exprs.push(PyExpr::from(arg.clone(), self.input_plan.clone()));
+                }
+                Ok(exprs)
+            }
```
Comment (andygrove) [inline dask_planner/src/expression.rs:309]: nit: functional is more concise here

```suggestion
            Expr::ScalarUDF { args, .. } => Ok(args
                .iter()
                .map(|arg| PyExpr::from(arg.clone(), self.input_plan.clone()))
                .collect()),
```

File: dask_planner/src/expression.rs [hunk 3]
```diff
@@ -321,6 +327,7 @@ impl PyExpr {
             Expr::Between { .. } => Ok("between".to_string()),
             Expr::Case { .. } => Ok("case".to_string()),
             Expr::IsNotNull(..) => Ok("is not null".to_string()),
+            Expr::ScalarUDF { fun, args: _ } => Ok(fun.name.clone()),
```
Comment (andygrove) [inline dask_planner/src/expression.rs:330]:
```suggestion
            Expr::ScalarUDF { fun, .. } => Ok(fun.name.clone()),
```

**Review by andygrove — APPROVED** [review]
LGTM. Left a couple of code suggestions.

**Pull request merged**